model/src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala

```diff
@@ -561,6 +568,36 @@ object WorkspaceState {
   case object DeleteFailed extends WorkspaceState
 }
 
+case class WorkspaceSetting(`type`: WorkspaceSettingType, config: WorkspaceSettingConfig)
+
+object WorkspaceSettingTypes {
+  sealed trait WorkspaceSettingType extends RawlsEnumeration[WorkspaceSettingType] {
+    override def toString: String = getClass.getSimpleName.stripSuffix("$")
+    override def withName(name: String): WorkspaceSettingType = WorkspaceSettingTypes.withName(name)
+    def defaultConfig(): WorkspaceSettingConfig
+  }
+
+  def withName(name: String): WorkspaceSettingType = name.toLowerCase match {
+    case "gcpbucketlifecycle" => GcpBucketLifecycle
+    case _                    => throw new RawlsException(s"invalid WorkspaceSetting [$name]")
+  }
+
+  case object GcpBucketLifecycle extends WorkspaceSettingType {
+    override def defaultConfig(): WorkspaceSettingConfig = GcpBucketLifecycleConfig(List.empty)
+  }
+}
+
+sealed trait WorkspaceSettingConfig
+object WorkspaceSettingConfig {
+  case class GcpBucketLifecycleConfig(rules: List[GcpBucketLifecycleRule]) extends WorkspaceSettingConfig
+
+  case class GcpBucketLifecycleRule(action: GcpBucketLifecycleAction, conditions: GcpBucketLifecycleCondition)
+  case class GcpBucketLifecycleAction(`type`: String)
+  case class GcpBucketLifecycleCondition(matchesPrefix: Set[String], age: Option[Int])
```

Comment:
I could add `type Days = Int` to the object so this would read as:

```
case class GcpBucketLifecycleCondition(matchesPrefix: Set[String], age: Option[Days])
```

Everything else would remain the same as it's functionally unchanged, but it would make it clearer to future devs